components/formats-gpl/src/loci/formats/in/ZeissCZIReader.java

```diff
@@ -963,6 +963,30 @@ else if (segment instanceof Attachment) {
       }
       indices.add(i);
       indexIntoPlanes.put(c, indices);
+      //Add series metadata : populate position list
+      for (DimensionEntry dimension : p.directoryEntry.dimensionEntries) {
+          if (dimension == null) {
+            continue;
+          }
+          switch (dimension.dimension.charAt(0)) {
+              case 'S':
+                  setSeries(p.coreIndex);
+                  String sIndex = "";
+                  int seriesId = p.coreIndex + 1;
+                  //add padding to make sure the original metadata table is organized properly in ImageJ
+                  if (seriesId < 10) {
+                      sIndex = "000" + String.valueOf(seriesId);
+                  }
+                  else if (seriesId >= 10 && seriesId < 100){
+                      sIndex = "00" + String.valueOf(seriesId);
+                  }
+                  else if (seriesId >= 100 && seriesId < 1000){
+                      sIndex = "0" + String.valueOf(seriesId);
+                  }
+                  addSeriesMetaList("Positions|" + "SeriesIndex|" + sIndex + "|", dimension.start);
+                  break;
+          }
+       }
```

Comment:
```setCoreIndex(0)``` should be added here, to restore the reader state to what it was before iterating over ```p.directory.dimensionEntries```.